lib/src/connector/connector.cc

```diff
@@ -129,16 +132,105 @@ void Connector::connect(int max_connect_attempts) {
             });
     }
 
+    // Check the state of the Connection instance, to avoid waiting
+    // for an Associate Session response in vain if a concurrent
+    // attempt to open a WebSocket connection is in progress as, in
+    // such case, Connection::connect() will not trigger the
+    // transmission of any Associate Session request.
+    // Note that we don't want to call associateSession()
+    // synchronously, as the broker would close the WebSocket if the
+    // session was already associated; such function must be invoked
+    // only by the WebSocket onOpen handler, asynchronously. So, first
+    // ensure that the connection is closed and then open it.
+    switch (connection_ptr_->getConnectionState()) {
+        case(ConnectionStateValues::connecting):
+        case(ConnectionStateValues::open):
+            LOG_DEBUG("There's an ongoing attempt to create a WebSocket connection; "
+                      "ensuring that it's closed before Associate Session");
+            try {
+                connection_ptr_->close(CloseCodeValues::normal,
+                                       "must Associate Session again");
+                LOG_TRACE("Waiting for the WebSocket connection to be closed, "
+                          "for a maximum of %1% s", WS_CONNECTION_CLOSE_TIMEOUT_S);
+                lth_util::Timer timer {};
+
+                while (connection_ptr_->getConnectionState()
+                            != ConnectionStateValues::closed
+                       && timer.elapsed_seconds() < WS_CONNECTION_CLOSE_TIMEOUT_S)
+                    Util::this_thread::sleep_for(Util::chrono::milliseconds(100));
```

Comment:
I don't like these semi busy loops, would it make sense to reimplement those using ```Util::condition_variable```?